### DISCUSSION TITLE
How to extend to a plugin option

### DISCUSSION THREAD

**Discussion by Eddio0141** [nix-community/nixvim#1262] [Q&A] [answered]
Hiya
I want to extend to the attribute in [`plugins.noice.cmdline.format`](https://nix-community.github.io/nixvim/plugins/noice/cmdline.html#pluginsnoicecmdlineformat) but I can't figure out how
I was finding an equivalent of `mkOptionDefault` but I can't find it after some looking

Any help?

**Comment by GaetanLepage** [comment]
Hey !
What do you mean by "extend" ?
Is the type of the option (`attrsOf anything`) incorrect ?

**Comment by Eddio0141** [comment]
So `plugins.noice.cmdline.format` has a default attribute
Instead of completely overriding it, I want to add or edit the default attribute

**Reply by GaetanLepage** [reply]
Ok, so if you want to override a specific attribute, you can do:
```nix
cmdline.format.search_down = {
  kind = "search";
  pattern = "_____";
  icon = "blabla";
  lang = "regex";
};
```
I think that in this case, the rest of the keys will keep their default values.
Maybe you could even override to the sub-attribute level, not sure.

**Reply by GaetanLepage** [reply, marked as answer]
I would expect that the table/attrs you provide is merged with the defaults...

**Reply by Eddio0141** [reply]
Ok I didn't try this for some reason but it seemed like that's the case, thanks!
